pyproject.toml

```diff
@@ -72,35 +71,35 @@ profile = "black"
 skip_glob = 'blackboxopt/examples/**'
 filter_files = true
 
-[tool.pylint.format]
-max-line-length = 88
-
-[tool.pylint.messages_control]
-generated-members = ["torch.*", "numpy.*"]
-disable = [
-    # Switched off due to possible collisions with black:
-    "bad-continuation",
-    "bad-whitespace",
-    # Switched off to be less aggressive
-    "missing-module-docstring",
-    "missing-class-docstring",
-    "missing-function-docstring",
-    "invalid-name",
-    "logging-not-lazy",
-    "logging-format-interpolation",
-    "logging-fstring-interpolation",
-    "duplicate-code",
-    "too-many-instance-attributes",
-    "too-many-arguments",
-    "too-few-public-methods",
-    "too-many-function-args"
+[tool.ruff]
+# Enable pycodestyle (`E`) and Pyflakes (`F`) codes by default.
+select = ["E", "F"]
+ignore = []
+# Allow autofix for all enabled rules (when `--fix`) is provided.
+fixable = ["A", "B", "C", "D", "E", "F"]
```

Comment:
How do you feel about adding `"I"` and drop `isort` accordingly?

It could also be useful to add `"RUF"` for applying [ruff specific rules](https://beta.ruff.rs/docs/rules/#ruff-specific-rules-ruf) (especially the `unused noqa` rule is quite useful).

Reply:
I'm happy to try it out :relaxed:

Reply:
Seems to work, although not 100% consistently with isort, which I don't think is an issue, though. I just noticed for the noqa annotated imports that isort converted into individual imports, ruff combined into one import again.
One downgrade in the current version as of 5c64453 is that I haven't figured out how to tell the ruff isort rules that for `examples/*` the line length is different. However, since this is not an issue right now, I'd propose to solve it when we need to. Unless you already know of a way?

Reply:
I'll move forward for now but would be happy to refine the setup as we go :relaxed: